core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -356,17 +358,17 @@ class LogCleanerTest {
     log.roll()
 
     // the first cleaning preserves the commit marker (at offset 3) since there were still records for the transaction
-    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), deleteHorizonMs = Long.MaxValue)._1
+    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), currentTime = largeDeleteHorizon)._1
     assertEquals(List(2, 1, 3), LogTest.keysInLog(log))
     assertEquals(List(3, 4, 5, 6, 7, 8), offsetsInLog(log))
 
     // delete horizon forced to 0 to verify marker is not removed early
-    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), deleteHorizonMs = 0L)._1
+    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), currentTime = 0L)._1
     assertEquals(List(2, 1, 3), LogTest.keysInLog(log))
     assertEquals(List(3, 4, 5, 6, 7, 8), offsetsInLog(log))
 
     // clean again with large delete horizon and verify the marker is removed
-    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), deleteHorizonMs = Long.MaxValue)._1
+    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), currentTime = largeDeleteHorizon)._1
```

Comment:
Why do we need to set current time to Long.MaxValue - tombstoneRetentionMs - 1? For verifying the removal of the tombstone, it's clearer if we set the currentTime in mockTime before the first round of cleaning and then explicit set current time to be tombstoneRetentionMs longer than that currentTime in a subsequent round of cleaning to verify that the tombstone is removed. Ditto below.

Reply:
This was one part of the test which I had some problems with. Notably, what happens is that we will try to calculate the delete horizon using Long.MaxValue as the current time. Inherently, an integer overflow error will occur (and we end up with some very low negative number). Therefore, I thought that we can get around it by setting the delete horizon to some value which  would not have problems with overflow (hence largeDeleteHorizon having the above value you mentioned.)

Reply:
Also, about setting mock time. Mock time is in fact never called in doClean. It is called in just clean(). The currentTime supplied to doClean is from clean(). So what you stated probably only applies to methods which call the regular clean() method.